src/ZEO/tests/test_marshal.py

```diff
@@ -0,0 +1,25 @@
+import unittest
+from ZEO.asyncio.marshal import encode
+from ZEO.asyncio.marshal import pickle_server_decode
+from ZopeUndo.Prefix import Prefix
+
+
+class MarshalTests(unittest.TestCase):
+
+    def testServerDecodeZopeUndoFilter(self):
+        # this is an example of Zope's 
+        args = (0, 20, {'user_name': Prefix('test')})
+        # test against repr because Prefix __eq__ operator
+        # doesn't compare Prefix with Prefix but only
+        # Prefix with strings. see Prefix.__doc__
+        self.assertEqual(
+            repr(pickle_server_decode(encode(*args))),
+            repr(args)
+        )
```

Comment:
How does this test exercise the `copy_reg` / `__bulitins__` bits being added as the "fix"?

Reply:
I added some information in the test comment.
Note: Probably the problem is only shown for the Zope version 2 branches, I see that on Zope4 ZopeUndo is completely eliminated (currently used by last Plone),

Reply:
afaik in Zope4 Undo is still there, but was moved into the control panel and isn't available on most ZMI tabs.

Reply:
@tseaver done